code/obj/item/gun/gun_parent.dm

```diff
@@ -175,7 +175,7 @@ var/list/forensic_IDs = new/list() //Global list of all guns, based on bioholder
 	..()
 	if(src.projectiles && length(src.projectiles) > 1)
 		src.current_projectile_num = ((src.current_projectile_num) % src.projectiles.len) + 1
-		src.set_current_projectile(src.projectiles[src.current_projectile_num])
+		src.set_current_projectile(src.projectiles[src.projectiles[src.current_projectile_num]])
```

Comment:
Er, what's this doing? Can you explain the types of the things here, because this looks nuts to me.

Reply:
src.projectiles is an associated list of projectile said name "execue", "bigshot", etc. to value which is the type of projectile itself.

so what the old code was doing was getting the key at some index, and setting that string as the projectile datum.
adding it once more gets the actual projectile instead of the string

Reply:
ah hold on im seeing this is weird in a different way. HUH